src/Lighthouses.java

```diff
@@ -0,0 +1,170 @@
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/* og�lnie je�li user wybierze pole z latarni� to nie mo�na mu na to pozwoli�, czyli je�li wstawi statek
+ * w pole z latarni� musi wyst�pi� wyj�tek (je�li pole do wstawienia statku b�dzie r�ne od '_'')
+ * trzeba b�dzie parstowa� zapisane pola na int �eby sprawdzi� ile statk�w o�wietl�
+ * trzeba zabroni� userowi wpisania innego znaku ni� int*/
+public class Lighthouses {
+
+    char board[][] = new char[12][12];
+    Scanner key = new Scanner(System.in);
+
+    public static void main( String[] args ) {
+        Scanner key = new Scanner(System.in);
+        Lighthouses lighthouses = new Lighthouses();
+        lighthouses.setBoard();
+        System.out.println("Tak wygl�da tablica do kt�rej trzeba wstawi� statki:");
+        System.out.println("'_' - to puste pole na kt�rym mo�na ustawi� statek.");
+        System.out.println("Cyfra w polu oznacza ile s�siednich p�l w pionie i poziomie o�wietla latarnia");
+        lighthouses.printBoard(lighthouses.board);
+        lighthouses.putShips(lighthouses.board);
+    }
+
+    public void printBoard( char board[][] ) {
```

Comment:
niepotrzebnie przekazujesz parametr board[][]. Parametry przekazujemy gdy "brakuje" nam danych. Musimy je wtedy przekazywać przez parametr.

Tzn. masz metodę która ma wyświetlić tablicę. Ale czy musisz w tym przypadku ją przekazać jako parametr ? Nie jest tak, że zawsze gdy chcę wyświetlić talbicę to muszę ją przekazać jako parametr. W tym konkretnym przypadku, słusznie dodałeś pole, które przechowuje całą planszę. Zasięg zmiennej (czyli też pola, inaczej zmiennej instancyjnej), jest widoczny w całej klasie, również wewnąrz metod (zagnieżdżanie klamer).
Tak wiec w metodzie printBoard masz dostęp do pola "board" nie musisz więc dodatkowo go przekazywać. Po prostu go wyświetl.